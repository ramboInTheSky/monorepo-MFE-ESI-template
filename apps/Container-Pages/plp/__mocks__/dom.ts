export function createElementWithChildren(totalChildren: number, amendChild?: (child: HTMLElement, i: number) => any) {
    const container = document.createElement("div")
    for (let i = 0; i < totalChildren; i += 1) {
        const child = document.createElement("div")
        jest.spyOn(child, "getBoundingClientRect")
        child.appendChild(document.createTextNode(`child ${i + 1}`))
        if (amendChild) amendChild(child, i)
        container.appendChild(child)
    }
    return container
}

export function mockWindowLocation(props = {}) {
    const {location} = window

    delete (window as any).location
    ;(window as any).location = {...location, ...props}

    const restore = () => {
        ;(window as any).location = location
    }

    return restore
}

export function mockWindowScrollTo() {
    const {scrollTo} = window
    delete (window as any).scrollTo
    window.scrollTo = jest.fn()
    return () => {
        window.scrollTo = scrollTo
    }
}
