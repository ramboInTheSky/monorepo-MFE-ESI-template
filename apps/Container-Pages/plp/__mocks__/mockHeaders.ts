import {TestDataSwitcher} from "../cypress/modules/filter"

export const amidoHeaders = TestDataSwitcher(
    ["postdeploy"],
    {
        "x-monorepo-language": "en",
        "x-monorepo-realm": "amido",
        "x-monorepo-territory": "gb",
        "x-set-caching": "false",
    },
    {
        "x-monorepo-language": "en",
        "x-monorepo-realm": "amido",
        "x-monorepo-territory": "gb",
        "test-with-local-esi": true,
    },
)
