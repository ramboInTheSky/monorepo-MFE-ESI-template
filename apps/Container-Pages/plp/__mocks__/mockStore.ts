import thunk from "redux-thunk"
import configureMockStore from "redux-mock-store"
import {ThemeColor, mockColors} from "../../../../libs/framework/themes/src"
import State from "../src/models/State"
import themeDimensions from "./themeDimensions.json"
import TextAlignment from "../src/models/textAlignment"
import {Dimensions} from "../src/models/Dimensions"
import {TextModel} from "../src/models/Text"
import {Theme} from "../src/models/theme"
import {SearchApiRequestTypes} from "../src/config/constants"
import {RequestDuckState} from "../src/ducks/request/types"
import text from "./default-text.json"

export const mockColours: ThemeColor = mockColors
export const mockDimensions: Dimensions = themeDimensions
export const mockText: TextModel = text

export const mockState: State = {
    text,
    textAlignment: TextAlignment.Ltr,
    request: {
        headers: {"x-monorepo-territory": "gb", "x-monorepo-language": "en"},
        url: "/",
        siteUrl: "",
        searchTerm: null,
        category: null,
        gender: null,
        territoryName: "",
        page: 6,
        type: SearchApiRequestTypes.Category,
        useDevEsi: false,
        viewportSize: "desktop",
        bloomreachPersonalizationEnabled: false,
    },
    search: {
        seoFilters: "",
        singleOptionFacetList: ["personalised", "feat", "sizetype", "deliverby"],
        searchBannerHtml: null,
        totalResults: 1234,
        filters: {
            Test1: {
                name: "Test1",
                facets: ["opt1"],
                displayName: "Test 1",
                type: "feat",
                isFilterOpen: true,
                isViewMoreOpen: true,
            },
            Test2: {
                name: "Test2 with options",
                facets: ["opt2"],
                displayName: "Test 2",
                type: "filter",
                isFilterOpen: false,
                isViewMoreOpen: false,
            },
            Test3: {
                name: "Test2 with options",
                displayName: "",
                type: "price",
                isFilterOpen: false,
                max: 100,
                min: 0,
                selectedMax: 70,
                selectedMin: 30,
                currencyCode: "GBP",
            },
            brand: {
                name: "Brand",
                facets: [
                    "brand:znth",
                    "brand:one",
                    "brand:oops",
                    "brand:two",
                    "brand:tcas",
                    "brand:three",
                    "brand:four",
                    "brand:five",
                    "brand:six",
                    "brand:2 size",
                ],
                displayName: "Test 1",
                type: "filter",
                isFilterOpen: true,
                isViewMoreOpen: true,
            },
            Price: {
                name: "Test2 with options",
                displayName: "",
                type: "price",
                isFilterOpen: false,
                max: 100,
                min: 0,
                selectedMax: 70,
                selectedMin: 30,
                currencyCode: "GBP",
            },
        },
        filtersSort: ["Test1", "Test2"],
        facets: {
            opt1: {n: "opt1", c: 1, v: "opt1", incompatibleWith: [], d: false},
            opt2: {n: "opt2", c: 1, v: "opt2", incompatibleWith: ["opt3"], d: false},
            opt3: {n: "opt3", c: 1, v: "opt3", incompatibleWith: ["opt2"], d: false},
        },
        items: [
            {itemNumber: "111111", newIn: true},
            {itemNumber: "222222", newIn: true},
        ],
        sorting: {
            selected: "score",
            options: [
                {
                    name: "Most Relevant",
                    value: "score",
                },
                {
                    name: "Most Popular",
                    value: "newpop",
                },
                {
                    name: "Alphabetical",
                    value: "alpha",
                },
                {
                    name: "Price: Low - High",
                    value: "pricelow",
                },
                {
                    name: "Price: High - Low",
                    value: "pricehi",
                },
            ],
        },
        startPage: 1,
        endPage: 1,
        isFetchingNextPage: false,
        isFetchingPreviousPage: false,
        isFetchingPageItems: false,
        debounceTime: 300,
        shouldFacetsBeFixed: false,
        autoCorrectQuery: null,
        title: "",
        initialItemsPerPage: 8,
        fetchTriggerOffset: {
            xs: 4,
            sm: 4,
            md: 6,
            lg: 6,
            xl: 8,
        },
        itemsPerRow: {
            inPageFiltersEnabled: {
                xs: 4,
                sm: 4,
                md: 6,
                lg: 6,
                xl: 8,
            },
            default: {
                xs: 4,
                sm: 4,
                md: 6,
                lg: 6,
                xl: 8,
            },
        },
        currentBreakpoint: "md",
        fetchTriggerOffsetRows: {
            xs: 2,
            sm: 2,
            md: 2,
            lg: 2,
            xl: 2,
        },
        subsequentPagesNonLazyloadRows: {
            xs: 2,
            sm: 2,
            md: 2,
            lg: 2,
            xl: 2,
        },
        itemsPerPage: {
            initial: {
                mobile: 8,
                tablet: 12,
                desktop: 24,
            },
            subsequent: {
                mobile: 10,
                tablet: 16,
                desktop: 28,
            },
        },
        relaxedQuery: "",
        includedComponents: [],
        searchCategory: {
            id: "",
            name: "",
        },
        bloomreachCookiesInitialVal: {
            brUid2: "",
            brMtSearch: "",
        },
    },
    viewAllModal: {
        displayName: "",
        name: "",
        isOpen: false,
        activeCharacter: "All",
        facets: {
            opt1: {n: "opt1", c: 1, v: "opt1", incompatibleWith: [], d: false},
            opt2: {n: "opt2", c: 1, v: "opt2", incompatibleWith: ["opt3"], d: false},
            opt3: {n: "opt3", c: 1, v: "opt3", incompatibleWith: ["opt2"], d: false},
        },
        hideSearchFilterModalElements: {
            letterNav: false,
            searchBox: false,
        },
    },
    seoFilters: {
        filterDepthCount: 6,
    },
    features: {
        enablePageInFilters: false,
        enableSearchBanners: false,
        enableTooltips: false,
    },
    categoryQuickLinks: {
        items: [
            {
                href: "/women",
                title: "Women",
                description: "Womens description...",
                imageSrc: "/some/image/src.jpg",
                imageAlt: "women",
            },
            {
                href: "/men",
                title: "Men",
                description: "Mens description...",
                imageSrc: "/some/image/src.jpg",
                imageAlt: "men",
            },
        ],
    },
    tabbedFilters: {
        isOpen: false,
        totalResults: 1234,
        filters: {
            Test1: {
                name: "Test1",
                facets: ["opt1"],
                displayName: "Test 1",
                type: "feat",
                isFilterOpen: true,
                isViewMoreOpen: true,
            },
            Test2: {
                name: "Test2 with options",
                facets: ["opt2"],
                displayName: "Test 2",
                type: "filter",
                isFilterOpen: false,
                isViewMoreOpen: false,
            },
            Test3: {
                name: "Test2 with options",
                displayName: "",
                type: "price",
                isFilterOpen: false,
                max: 100,
                min: 0,
                selectedMax: 70,
                selectedMin: 30,
                currencyCode: "GBP",
            },
            brand: {
                name: "Brand",
                facets: [
                    "brand:znth",
                    "brand:one",
                    "brand:oops",
                    "brand:two",
                    "brand:tcas",
                    "brand:three",
                    "brand:four",
                    "brand:five",
                    "brand:six",
                    "brand:2 size",
                ],
                displayName: "Test 1",
                type: "filter",
                isFilterOpen: true,
                isViewMoreOpen: true,
            },
            Price: {
                name: "Test2 with options",
                displayName: "",
                type: "price",
                isFilterOpen: false,
                max: 100,
                min: 0,
                selectedMax: 70,
                selectedMin: 30,
                currencyCode: "GBP",
            },
        },
        filtersSort: ["Test1", "Test2"],
        facets: {
            opt1: {n: "opt1", c: 1, v: "opt1", incompatibleWith: [], d: false},
            opt2: {n: "opt2", c: 1, v: "opt2", incompatibleWith: ["opt3"], d: false},
            opt3: {n: "opt3", c: 1, v: "opt3", incompatibleWith: ["opt2"], d: false},
        },
        items: [
            {itemNumber: "111111", newIn: true},
            {itemNumber: "222222", newIn: true},
        ],
        selectedFilter: null,
        brandSearch: "",
        historicFacetFilter: {},
        isPerformingKeyFilterSearch: false,
        canSync: true,
        isPriceFacetChanged: false,
    },
}

export const mockTextAlignment = {Value: "ltr"}

export const mockTheme: Theme = {colours: mockColours, dimensions: mockDimensions, direction: "ltr"}
export const mockConfigureStore = configureMockStore([thunk])
export const createMockStoreForRequest = (request: Partial<RequestDuckState>) => {
    return mockConfigureStore({...mockState, request: {...mockState.request, ...request}})
}
const mockStore = mockConfigureStore(mockState)

export default mockStore
