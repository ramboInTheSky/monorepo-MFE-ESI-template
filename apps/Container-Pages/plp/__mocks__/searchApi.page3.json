{
  "totalResults": 96,
  "filters": [
    {
      "name": "Feat",
      "displayName": "",
      "type": "feat",
      "options": [
        {
          "n": "New In",
          "c": 364,
          "v": "newin",
          "s": true
        },
        {
          "n": "In Stock",
          "c": 2405,
          "v": "instock"
        }
      ]
    },
    {
      "name": "Size",
      "displayName": "Size",
      "type": "filter",
      "options": [
        {
          "n": "1 Tall",
          "c": 24,
          "v": "1tall",
          "s": true
        },
        {
          "n": "2",
          "c": 3,
          "v": "2",
          "s": true
        },
        {
          "n": "3",
          "c": 3,
          "v": "3",
          "s": true
        },
        {
          "n": "4 Tall",
          "c": 24,
          "v": "4tall",
          "s": true
        },
        {
          "n": "5",
          "c": 3,
          "v": "5",
          "s": true
        },
        {
          "n": "6 Tall",
          "c": 24,
          "v": "6tall",
          "s": true
        },
        {
          "n": "7",
          "c": 3,
          "v": "7",
          "s": true
        },
        {
          "n": "8 Tall",
          "c": 24,
          "v": "8tall",
          "s": true
        },
        {
          "n": "9",
          "c": 3,
          "v": "9",
          "s": true
        }
      ]
    },
    {
      "name": "Colour",
      "displayName": "Display Colour",
      "type": "filter",
      "options": [
        {
          "n": "8 Tall",
          "c": 24,
          "v": "8tall",
          "s": true
        },
        {
          "n": "18",
          "c": 3,
          "v": "18",
          "s": true
        }
      ]
    },
    {
      "name": "Style",
      "displayName": "Style",
      "type": "filter",
      "options": [
        {
          "n": "Aaaaa Aaaaaa",
          "c": 24,
          "v": "1tall",
          "s": true
        },
        {
          "n": "Bbbbbbbbbb",
          "c": 3,
          "v": "2",
          "s": true
        },
        {
          "n": "Dddddd Dddddd DDddddddddddd",
          "c": 3,
          "v": "3"
        },
        {
          "n": "CCCCC",
          "c": 24,
          "v": "4tall",
          "s": true
        },
        {
          "n": "JJJJJJ",
          "c": 3,
          "v": "5"
        },
        {
          "n": "VVVVV",
          "c": 24,
          "v": "6tall",
          "s": true
        },
        {
          "n": "AAAA",
          "c": 3,
          "v": "7"
        },
        {
          "n": "XXXXX",
          "c": 24,
          "v": "8tall",
          "s": true
        },
        {
          "n": "9",
          "c": 3,
          "v": "9",
          "s": true
        },
        {
          "n": "10",
          "c": 3,
          "v": "10",
          "s": true
        },
        {
          "n": "11",
          "c": 3,
          "v": "11",
          "s": true
        },
        {
          "n": "12",
          "c": 3,
          "v": "12",
          "s": true
        },
        {
          "n": "13",
          "c": 3,
          "v": "13",
          "s": true
        },
        {
          "n": "14",
          "c": 3,
          "v": "14",
          "s": true
        },
        {
          "n": "15",
          "c": 3,
          "v": "15",
          "s": true
        },
        {
          "n": "16",
          "c": 3,
          "v": "16",
          "s": true
        },
        {
          "n": "17",
          "c": 3,
          "v": "17",
          "s": true
        },
        {
          "n": "18",
          "c": 3,
          "v": "18",
          "s": true
        },
        {
          "n": "19",
          "c": 3,
          "v": "19",
          "s": true
        },
        {
          "n": "20",
          "c": 3,
          "v": "20",
          "s": true
        },
        {
          "n": "21",
          "c": 3,
          "v": "21",
          "s": true
        }
      ]
    },
    {
      "name": "Price",
      "displayName": "Price",
      "type": "price",
      "currencyCode": "GBP",
      "min": 0,
      "max": 100,
      "selectedMin": 20,
      "selectedMax": 100
    }
  ],
  "items": [
    {
      "itemNumber": "X3999999",
      "newIn": true
    },
    {
      "itemNumber": "X3800845",
      "newIn": false
    },
    {
      "itemNumber": "X3111111",
      "newIn": false
    },
    {
      "itemNumber": "X32222",
      "newIn": false
    },
    {
      "itemNumber": "X3333",
      "newIn": true
    },
    {
      "itemNumber": "X34444",
      "newIn": false
    },
    {
      "itemNumber": "X35555",
      "newIn": false
    },
    {
      "itemNumber": "X36666",
      "newIn": false
    },
    {
      "itemNumber": "X380077784",
      "newIn": false
    },
    {
      "itemNumber": "X31008884",
      "newIn": true
    },
    {
      "itemNumber": "X32008884",
      "newIn": true
    },
    {
      "itemNumber": "X33008884",
      "newIn": true
    },
    {
      "itemNumber": "X34008884",
      "newIn": true
    },
    {
      "itemNumber": "X35008884",
      "newIn": true
    },
    {
      "itemNumber": "X36008884",
      "newIn": true
    },
    {
      "itemNumber": "X37008884",
      "newIn": true
    },
    {
      "itemNumber": "X37108884",
      "newIn": true
    },
    {
      "itemNumber": "X37208884",
      "newIn": true
    },
    {
      "itemNumber": "X37308884",
      "newIn": true
    },
    {
      "itemNumber": "X37408884",
      "newIn": true
    },
    {
      "itemNumber": "X37508884",
      "newIn": true
    },
    {
      "itemNumber": "X37608884",
      "newIn": true
    },
    {
      "itemNumber": "X37808884",
      "newIn": true
    },
    {
      "itemNumber": "X37908884",
      "newIn": true
    }
  ],
  "pagination": {
    "totalResults": 2403,
    "numberOfItems": 24,
    "start": 0,
    "previousPageUrl": "http://resourceUrl/resourceEndpoint?params",
    "nextPageUrl": "http://resourceUrl/resourceEndpoint?params"
  },
  "keywordRedirect": {
    "redirectedUrl": "/clearance",
    "redirectedQuery": "clearance",
    "originalQuery": "clearance"
  },
  "sorting": {
    "selected": "newpop",
    "options": [
      {
        "name": "Most Relevant",
        "value": "score"
      },
      {
        "name": "Most Popular",
        "value": "newpop"
      },
      {
        "name": "Alphabetical",
        "value": "alpha"
      },
      {
        "name": "Price: Low - High",
        "value": "pricelow"
      },
      {
        "name": "Price: High - Low",
        "value": "pricehi"
      }
    ]
  },
  "redirect": null
}
