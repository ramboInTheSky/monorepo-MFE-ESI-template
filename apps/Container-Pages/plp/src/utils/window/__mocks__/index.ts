import {
    getWindow,
    getDocumentScrollTop,
    removeDocumentScrollListener,
    addDocumentScrollListener,
    isSomewhatInViewport,
    isAboveViewport,
    isBelowViewport,
    IS_BROWSER,
} from ".."

jest.mock("..", () => ({
    getWindow: jest.fn(() => ({})),
    IS_BROWSER: jest.fn(() => true),
    isAboveViewport: jest.fn(),
    isBelowViewport: jest.fn(),
    isSomewhatInViewport: jest.fn(),
    getDocumentScrollTop: jest.fn(),
    addDocumentScrollListener: jest.fn(),
    removeDocumentScrollListener: jest.fn(),
}))

function mockListener(fn: jest.Mock) {
    const api = {
        listener: () => null,
    }
    fn.mockImplementation(listener => {
        api.listener = listener
    })
    return api
}

export function mockGetWindow(value: any = window) {
    ;(getWindow as jest.Mock).mockReturnValue(value)
}

export function mockGetDocumentScrollTop(value: number) {
    ;(getDocumentScrollTop as jest.Mock).mockReturnValue(value)
}

export function mockRemoveDocumentScrollListener() {
    return mockListener(removeDocumentScrollListener as jest.Mock)
}

export function mockAddDocumentScrollListener() {
    return mockListener(addDocumentScrollListener as jest.Mock)
}

export function mockIsSomewhatInViewport(result: boolean) {
    ;(isSomewhatInViewport as jest.Mock).mockReturnValue(result)
}

export function mockIsAboveViewport(result: boolean) {
    ;(isAboveViewport as jest.Mock).mockReturnValue(result)
}

export function mockIsBelowViewport(result: boolean) {
    ;(isBelowViewport as jest.Mock).mockReturnValue(result)
}

export function mockIsBrowser(result: boolean) {
    ;(IS_BROWSER as jest.Mock).mockReturnValue(result)
}
