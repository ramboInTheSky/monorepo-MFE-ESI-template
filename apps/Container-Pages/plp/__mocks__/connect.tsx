import React from "react"
import {Provider} from "react-redux"
import {render} from "@testing-library/react"
import {mockState, mockConfigureStore} from "./mockStore"

interface AnyObject {
    [key: string]: any
}

interface MockComponentProps {
    propsRef: {current: any}
    props: AnyObject
}

const MockComponent = jest.fn(({propsRef, ...props}: MockComponentProps) => {
    // eslint-disable-next-line no-param-reassign
    propsRef.current = props
    return null
}) as any

export function renderConnect(connect: Function, state: AnyObject = {}, otherProps: Record<any, any> = {}) {
    const props = React.createRef()
    const store = mockConfigureStore({...mockState, ...state})
    const ConnectedMock = connect(MockComponent)
    render(
        <Provider store={store}>
            <ConnectedMock {...otherProps} propsRef={props} />
        </Provider>,
    )
    return props.current
}

export function createMockStateForTabbedFilters(searchState: Record<any, any> = {}) {
    return {
        ...mockState,
        tabbedFilters: {
            ...mockState.tabbedFilters,
            ...searchState,
        },
    }
}
