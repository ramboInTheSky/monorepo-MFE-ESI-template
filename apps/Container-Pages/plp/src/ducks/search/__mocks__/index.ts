import {mockState} from "../../../../__mocks__/mockStore"

import {selectHasNextPage, selectHasPreviousPage} from ".."

import {selectFetchTriggerOffset, selectSubsequentItemsPerPage} from "../utils"

jest.mock("../utils", () => ({
    selectFetchTriggerOffset: jest.fn(),
    selectSubsequentItemsPerPage: jest.fn(),
}))
jest.mock("..", () => ({
    fixFacets: jest.fn(() => ({type: "FIX_FACETS"})),
    unfixFacets: jest.fn(() => ({type: "UNFIX_FACETS"})),
    selectItems: jest.fn(),
    selectHasNextPage: jest.fn(),
    selectHasPreviousPage: jest.fn(),
    selectCurrentEndPage: jest.fn(),
    selectCurrentStartPage: jest.fn(),
    selectTotalItemsPerPage: jest.fn(),
    fetchNextPageItemsThunk: jest.fn(() => ({type: "foo"})),
    fetchPreviousPageItemsThunk: jest.fn(() => ({type: "foo"})),
}))

export function mockSelectHasNextPage(result) {
    ;(selectHasNextPage as jest.Mock).mockReturnValue(result)
}

export function mockSelectHasPreviousPage(result) {
    ;(selectHasPreviousPage as jest.Mock).mockReturnValue(result)
}

export function mockSelectFetchTriggerOffset(result) {
    ;(selectFetchTriggerOffset as jest.Mock).mockReturnValue(result)
}
export function mockSelectSubsequentItemsPerPage(result) {
    ;(selectSubsequentItemsPerPage as jest.Mock).mockReturnValue(result)
}

export function createMockStateForSearch(searchState: Record<any, any> = {}) {
    return {
        ...mockState,
        search: {
            ...mockState.search,
            ...searchState,
        },
    }
}
