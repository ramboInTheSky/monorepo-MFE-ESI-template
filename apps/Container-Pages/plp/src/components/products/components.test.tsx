import React from "react"
import {render} from "@testing-library/react"
import {ThemeProvider} from "styled-components"
import {mockTheme} from "../../../__mocks__/mockStore"
import {
    ProductGrid,
    ProductsRoot,
    ProductGridItem,
    ProductGridWrapper
} from "./components"

describe("Products components -", () => {
    test("ProductsGridContainer", () => {
        const {asFragment} = render(
            <ThemeProvider theme={mockTheme}>
                <ProductsRoot>Test</ProductsRoot>
            </ThemeProvider>,
        )
        expect(asFragment()).toMatchSnapshot()
    })
    test("ProductGridWrapper", () => {
        const {asFragment} = render(
            <ThemeProvider theme={mockTheme}>
                <ProductGridWrapper>Test</ProductGridWrapper>
            </ThemeProvider>,
        )
        expect(asFragment()).toMatchSnapshot()
    })
    test("ProductGridWrapper with larger header", () => {
        const {asFragment} = render(
            <ThemeProvider theme={mockTheme}>
                <ProductGridWrapper $largerTopPaddingDesktop>Test</ProductGridWrapper>
            </ThemeProvider>,
        )
        expect(asFragment()).toMatchSnapshot()
    })

    test("ProductGrid", () => {
        const {asFragment} = render(<ProductGrid>Test</ProductGrid>)
        expect(asFragment()).toMatchSnapshot()
    })

    test("ProductGridItem", () => {
        const {asFragment} = render(
            <ThemeProvider theme={mockTheme}>
                <ProductGridItem>Test</ProductGridItem>
            </ThemeProvider>,
        )
        expect(asFragment()).toMatchSnapshot()
    })

    test("ProductGridItem when", () => {
        const {asFragment} = render(
            <ThemeProvider theme={mockTheme}>
                <ProductGridItem>Test</ProductGridItem>
            </ThemeProvider>,
        )
        expect(asFragment()).toMatchSnapshot()
    })
})
