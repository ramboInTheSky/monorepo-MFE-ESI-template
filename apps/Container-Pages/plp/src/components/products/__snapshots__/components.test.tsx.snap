// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products components - ProductGrid 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  z-index: 0;
}

<div
    class="MuiGrid-root c0 MuiGrid-container"
  >
    Test
  </div>
</DocumentFragment>
`;

exports[`Products components - ProductGridItem 1`] = `
<DocumentFragment>
  .c0 {
  padding: 0 1rem 0 1rem;
}

@media (max-width:767px) {
  .c0 {
    padding: 0 0.5rem 0 0.5rem;
  }
}

<div
    class="MuiGrid-root c0 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
  >
    Test
  </div>
</DocumentFragment>
`;

exports[`Products components - ProductGridItem when 1`] = `
<DocumentFragment>
  .c0 {
  padding: 0 1rem 0 1rem;
}

@media (max-width:767px) {
  .c0 {
    padding: 0 0.5rem 0 0.5rem;
  }
}

<div
    class="MuiGrid-root c0 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
  >
    Test
  </div>
</DocumentFragment>
`;

exports[`Products components - ProductGridWrapper 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  -webkit-flex: 1 0 80%;
  -ms-flex: 1 0 80%;
  flex: 1 0 80%;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

@media (min-width:1024px) {
  .c0 {
    -webkit-flex: 1 0 25%;
    -ms-flex: 1 0 25%;
    flex: 1 0 25%;
    padding-top: 4rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding-left: 0rem;
    padding-right: 0rem;
  }
}

@media (min-width:1024px) {
  .c0 {
    -webkit-flex: 1 0 75%;
    -ms-flex: 1 0 75%;
    flex: 1 0 75%;
  }
}

@media (min-width:1280px) {
  .c0 {
    -webkit-flex: 1 0 80%;
    -ms-flex: 1 0 80%;
    flex: 1 0 80%;
  }
}

<div
    class="MuiGrid-root c0 MuiGrid-container"
  >
    Test
  </div>
</DocumentFragment>
`;

exports[`Products components - ProductGridWrapper with larger header 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  -webkit-flex: 1 0 80%;
  -ms-flex: 1 0 80%;
  flex: 1 0 80%;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

@media (min-width:1024px) {
  .c0 {
    -webkit-flex: 1 0 25%;
    -ms-flex: 1 0 25%;
    flex: 1 0 25%;
    padding-top: 4.25rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding-left: 0rem;
    padding-right: 0rem;
  }
}

@media (min-width:1024px) {
  .c0 {
    -webkit-flex: 1 0 75%;
    -ms-flex: 1 0 75%;
    flex: 1 0 75%;
  }
}

@media (min-width:1280px) {
  .c0 {
    -webkit-flex: 1 0 80%;
    -ms-flex: 1 0 80%;
    flex: 1 0 80%;
  }
}

<div
    class="MuiGrid-root c0 MuiGrid-container"
  >
    Test
  </div>
</DocumentFragment>
`;

exports[`Products components - ProductsGridContainer 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  -webkit-flex: 1 0 80%;
  -ms-flex: 1 0 80%;
  flex: 1 0 80%;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  min-width: 0;
}

@media (min-width:1024px) {
  .c0 {
    -webkit-flex: 1 0 75%;
    -ms-flex: 1 0 75%;
    flex: 1 0 75%;
  }
}

@media (min-width:1024px) {
  .c0 {
    -webkit-flex: 1 0 75%;
    -ms-flex: 1 0 75%;
    flex: 1 0 75%;
  }
}

@media (min-width:1280px) {
  .c0 {
    -webkit-flex: 1 0 80%;
    -ms-flex: 1 0 80%;
    flex: 1 0 80%;
  }
}

<div
    class="MuiGrid-root c0 MuiGrid-container"
  >
    Test
  </div>
</DocumentFragment>
`;
